src/v/archival/ntp_archiver_service.cc

```diff
@@ -2248,6 +2248,15 @@ ss::future<> ntp_archiver::apply_retention() {
         co_return;
     }
 
+    if (manifest().archive_size_bytes() != 0) {
```

Comment:
Looking at the condition right before this one, is it possible that archive and STM start offsets are not equal and `arch_so == model::offset{}` so we end up here?